### PR TITLE
fix(Input): changing the default styles for Input component

### PR DESCRIPTION
**Pull request by alinais**
# Input

Updating the default classes for the input to match the light view of the Teams UI.

### TODO

- [x] Conformance test
- [x] Minimal doc site example
- [x] Stardust base theme
- [x] Teams Light theme
- [ ] Teams Dark theme
- [ ] Teams Contrast theme
- [ ] Confirm RTL usage
- [x] [W3 accessibility](https://www.w3.org/standards/webdesign/accessibility) check
- [ ] [Stardust accessibility](https://github.com/stardust-ui/accessibility) check
- [ ] Update glossary props table
- [ ] Update the CHANGELOG.md

# UI Update

Simple input
image

Simple input focused
image

Input variation with icon
image

### CODE REVIEW

**Comment by codepretty** [comment]
I don't see a way to change the width of the input. Is that going to be added later?

**Comment by codepretty** [comment]
Looks like the bottom focus border angle is inverted. It should be angled up at the corners instead of down...

image

File: src/components/Input/inputRules.ts [hunk 1]
```diff
@@ -9,13 +9,22 @@ const inputRules = {
       borderRadius: variables.borderRadius,
       outline: 0,
       padding: variables.defaultPadding,
+      backgroundColor: variables.backgroundColor,
+      ':focus-within': {
```
[the remark this answers is not in the document]
Comment (alinais) [reply src/components/Input/inputRules.ts]: Thanks for the help. I changed the approach to move input at the full size of the parent div and basically the focus will stay attached to the input, in a classic way that works in Edge

File: src/components/Input/Input.tsx [hunk 2]
```diff
@@ -81,11 +92,18 @@ class Input extends UIComponent<any, any> {
...
+      tabIndex: this.computeTabIndex,
+    }
+  }
+
   renderComponent({ ElementType, classes, rest }) {
     const { children, className, icon, input, type } = this.props
     const [htmlInputProps, restProps] = this.partitionProps()
 
     const inputClasses = cx(classes.input)
+    const iconClasses = cx(classes.icon)
```
Comment (mnajdova) [inline src/components/Input/Input.tsx]: No need for using cx here, as the classes are already calculated.

File: src/components/Input/Input.tsx [hunk 3]
```diff
@@ -108,9 +126,15 @@ class Input extends UIComponent<any, any> {
         <ElementType {...rest} className={classes.root} {...htmlInputProps}>
           {createHTMLInput(input || type, {
             defaultProps: htmlInputProps,
```
Comment (mnajdova) [inline src/components/Input/Input.tsx]: We can extract the creation of the input in a separate variable and reuse it in the both return statements. I would actually refactor this with the following:
```jsx
return (<ElementType {...rest} className={classes.root} {...htmlInputProps}>
          {createHTMLInput(input || type, {
            defaultProps: htmlInputProps,
            overrideProps: {
              className: inputClasses,
              ref: this.handleInputRef,
            },
          })}
          {this.computeIcon() && Icon.create(this.computeIcon(), {
            defaultProps: { className: iconClasses },
            overrideProps: predefinedProps => this.handleIconOverrides,
          })}
        </ElementType>)
```

**Comment by codepretty** [comment]
If you have time, we should update the doc to just be "an icon" instead of "a search icon" since it really could be anything there.
image

File: src/components/Input/Input.tsx (hunk 2 above)
Comment (alinais) [reply src/components/Input/Input.tsx]: fixed

File: src/components/Input/Input.tsx (hunk 3 above)
Comment (alinais) [reply src/components/Input/Input.tsx]: Thanks. Fixed.

**Comment by alinais** [comment]
@codepretty , thanks. Fixed.

**Pull request closed without merging**